### ISSUE TITLE
Not working with CUDNN 8

### ISSUE THREAD

**Issue by MatthijsBurgh** [CMU-Perceptual-Computing-Lab/openpose#1913]
### Issue Summary
Issues with Caffe and CUDNN 8, when compiling without CUDNN, the problem doesn't show.

```bash
/build/examples/openpose/openpose.bin --video examples/media/video.avi --hand --net_resolution 80x32
Starting OpenPose demo...
Configuring OpenPose...
Starting thread(s)...
Auto-detecting all available GPUs... Detected 1 GPU(s), using 1 of them starting at GPU 0.
F0406 11:35:36.460844 73660 cudnn_conv_layer.cpp:53] Check failed: status == CUDNN_STATUS_SUCCESS (1 vs. 0)  CUDNN_STATUS_NOT_INITIALIZED
*** Check failure stack trace: ***
    @     0x7f888dd321c3  google::LogMessage::Fail()
    @     0x7f888dd3725b  google::LogMessage::SendToLog()
    @     0x7f888dd31ebf  google::LogMessage::Flush()
    @     0x7f888dd326ef  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f888d95d7d5  caffe::CuDNNConvolutionLayer<>::LayerSetUp()
    @     0x7f888da41e55  caffe::Net<>::Init()
    @     0x7f888da44345  caffe::Net<>::Net()
    @     0x7f888e40be9c  op::NetCaffe::initializationOnThread()
    @     0x7f888e425aa5  op::addCaffeNetOnThread()
    @     0x7f888e426c9d  op::PoseExtractorCaffe::netInitializationOnThread()
    @     0x7f888e42b585  op::PoseExtractorNet::initializationOnThread()
    @     0x7f888e423477  op::PoseExtractor::initializationOnThread()
    @     0x7f888e41ed07  op::WPoseExtractor<>::initializationOnThread()
    @     0x7f888e451c57  op::Worker<>::initializationOnThreadNoException()
    @     0x7f888e451da8  op::SubThread<>::initializationOnThread()
    @     0x7f888e4535c8  op::Thread<>::initializationOnThread()
    @     0x7f888e456c0c  op::Thread<>::threadFunction()
    @     0x7f888e071d84  (unknown)
    @     0x7f888dd72609  start_thread
    @     0x7f888deb0293  clone
Aborted (core dumped)
```

### Executed Command (if any)
```bash
/build/examples/openpose/openpose.bin --video examples/media/video.avi --hand --net_resolution 80x32 --logging_level 0 --disable_multi_thread
```
### OpenPose Output (if any)
```bash
Starting OpenPose demo...
Configuring OpenPose...
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToProducer():210
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToProducerType():170
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToPoseMode():11
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToPoseModel():33
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToScaleMode():82
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToHeatMapScaleMode():116
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToDetector():147
/home/username/src/openpose/src/openpose/utilities/flagsToOpenPose.cpp:flagsToDetector():147
Starting thread(s)...
Running configureThreadManager...
/home/username/src/openpose/src/openpose/producer/producer.cpp:createProducer():417
renderModePose = 3
renderModeFace = 3
renderModeHand = 3
renderOutput = 1
renderOutputGpu = 1
renderFace = 0
renderHand = 1
renderHandGpu = 1
/home/username/src/openpose/src/openpose/wrapper/wrapperAuxiliary.cpp:wrapperConfigureSanityChecks():17
/home/username/src/openpose/src/openpose/wrapper/wrapperAuxiliary.cpp:wrapperConfigureSanityChecks():192
userInputAndPreprocessingWsEmpty = 1
userOutputWsEmpty = 1
numberGpuThreads = -1
gpuNumberStart = 0
Auto-detecting all available GPUs... Detected 1 GPU(s), using 1 of them starting at GPU 0.
writeImagesCleaned = 
writeKeypointCleaned = 
writeJsonCleaned = 
writeHeatMapsCleaned = 
modelFolder = models/
finalOutputSize = [1280,720]
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():370
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():435
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():440
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():451
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():505
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():510
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():557
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():601
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():606
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():645
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():662
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():679
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():698
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():714
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():747
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():772
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():781
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():797
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():798
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():807
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():824
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():844
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():865
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():874
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():878
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():884
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():943
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():959
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():965
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():1041
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():1075
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():1179
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():1203
/home/username/src/openpose/include/openpose/wrapper/wrapperAuxiliary.hpp:configureThreadManager():1210
/home/username/src/openpose/include/openpose/wrapper/wrapper.hpp:exec():419
/home/username/src/openpose/include/openpose/thread/threadManager.hpp:exec():185
/home/username/src/openpose/include/openpose/thread/queueBase.hpp:addPusher():362
/home/username/src/openpose/include/openpose/thread/queueBase.hpp:addPusher():362
/home/username/src/openpose/include/openpose/thread/queueBase.hpp:addPusher():362
/home/username/src/openpose/include/openpose/thread/threadManager.hpp:exec():190
/home/username/src/openpose/include/openpose/thread/thread.hpp:threadFunction():182
/home/username/src/openpose/include/openpose/thread/thread.hpp:initializationOnThread():167
Starting initialization on thread. In /home/username/src/openpose/src/openpose/pose/poseExtractorCaffe.cpp:netInitializationOnThread():172
F0406 11:37:32.577636 73782 cudnn_conv_layer.cpp:53] Check failed: status == CUDNN_STATUS_SUCCESS (1 vs. 0)  CUDNN_STATUS_NOT_INITIALIZED
*** Check failure stack trace: ***
    @     0x7fa9bb1e11c3  google::LogMessage::Fail()
    @     0x7fa9bb1e625b  google::LogMessage::SendToLog()
    @     0x7fa9bb1e0ebf  google::LogMessage::Flush()
    @     0x7fa9bb1e16ef  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fa9bae0c7d5  caffe::CuDNNConvolutionLayer<>::LayerSetUp()
    @     0x7fa9baef0e55  caffe::Net<>::Init()
    @     0x7fa9baef3345  caffe::Net<>::Net()
    @     0x7fa9bb8bae9c  op::NetCaffe::initializationOnThread()
    @     0x7fa9bb8d4aa5  op::addCaffeNetOnThread()
    @     0x7fa9bb8d5c9d  op::PoseExtractorCaffe::netInitializationOnThread()
    @     0x7fa9bb8da585  op::PoseExtractorNet::initializationOnThread()
    @     0x7fa9bb8d2477  op::PoseExtractor::initializationOnThread()
    @     0x7fa9bb8cdd07  op::WPoseExtractor<>::initializationOnThread()
    @     0x7fa9bb900c57  op::Worker<>::initializationOnThreadNoException()
    @     0x7fa9bb900da8  op::SubThread<>::initializationOnThread()
    @     0x7fa9bb9025c8  op::Thread<>::initializationOnThread()
    @     0x7fa9bb905c0c  op::Thread<>::threadFunction()
    @     0x7fa9bb906065  op::Thread<>::exec()
    @     0x7fa9bb908d44  op::ThreadManager<>::exec()
    @     0x7fa9bb961862  op::WrapperT<>::exec()
    @     0x55d91607320c  openPoseDemo()
    @     0x55d91606f6e7  main
    @     0x7fa9bb2640b3  __libc_start_main
    @     0x55d91606f71e  _start
Aborted (core dumped)
```

### Type of Issue
Select the topic(s) on your post, delete the rest:
- Execution error

### Your System Configuration
2. **OpenPose version**: master (1f1aa9c59fe59c90cca685b724f4f97f76137224)

3. **General configuration**:
    - **Installation mode**: CMake
    - **Operating system**: Ubuntu
    - **Operating system version** (e.g., Ubuntu 16, Windows 10, ...): Ubuntu 20.04
    - **Release or Debug mode**? (by default: release): release
    - Compiler: gcc 9.3.0

4. **Non-default settings**:
    - **3-D Reconstruction module added**? (by default: no): no
    - Any other custom CMake configuration with respect to the default version? (by default: no): cmake .. -DBUILD_PYTHON=ON -DPYTHON_EXECUTABLE=/usr/bin/python3.8 -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.8.so

5. **3rd-party software**:
    - **Caffe version**: Default from OpenPose
    - **CMake version**: 3.16.3
    - **OpenCV version**: apt-get version 4.2.0

6. If **GPU mode** issue:
    - **CUDA version**: 11.0.228
    - **cuDNN version**: 8.0.5
    - **GPU model**: Quadro M1200, Driver 460.39

**Comment by XFFer** [comment]
Give up bro! I attempted many times for three days and nights and got a bad cold. It was a fucking code. 
It cannot support Cudnn8 and CUDA>=11.0. 
I tried to modify all the config files even the source code of Caffe and OpenPose 1.7.0. 
Because Caffe cannot support Cudnn>=8, I eventually cannot run the binary file and have the same error.
Damn it! Fuck!
The developer said the problem can be solved by update the NVIDIA driver or turn to use the CUDA10+Cudnn7. It was a fucking solution. I tried and tried and tried until Ubuntu cannot open with a black screen. I reinstall it and don't wanna try again. Fuck!

**Comment by MatthijsBurgh** [comment]
I do not experience it on all PCs I use. I have one with a GTX1080 (mobile). That one runs fine. So probably the issue is deep inside cuda/cudnn. Somewhere only Nvidia can fix it. So I think this one can't be solved by us. So I am closing it.

(P.S. @XFFer please mind your language a little bit)

**Comment by SpongeBab** [comment]
@XFFer OH! Damn it. I tried and tried and tried.My cuda version is cuda 11.0. I have checked various solutions and there is no installation tutorial for cuda11.0. Occasionally I saw that caffe has been deprecated. I just realized that the new version of cuda is not supported. I must use cuda10.1 in 2021, really speechless.

**Comment by italosalgado14** [comment]
@SpongeBab Were you able to run it? I have an Nvidia 3070 card and caffe apparently does not recognize cudnn 8, because of the ampere architecture we are forced to use cuda 11. I can't run openpose then on a modern architecture? :(

**Comment by VisionaryMind** [comment]
I spent a couple of days running this issue through multiple tests on two separate GPUs, one NVIDIA 2080 with 8Gb of RAM and another witih 16Gb. The conclusion I have come to is that OpenPpose simply doesn't support the latest NVIDIA cards. Originally, I started with CUDA 11.5 / cuDNN 8.3, and piped the output through a custom script that measures GPU RAM usage. OpenPose, with or without hand and face detection, will consume the entire 8Gb of RAM within the first 500 milliseconds of any operation, even if there is only one 640x480 image used as input. It happens so quickly, that my first impression is that there is no memory management in the code. Even after creating a virtual env with CUDA 10.2 / cuDNN 7.6, the same problem is seen. Hopefully, the OpenPose developers will consider updating code to work with newer CUDA versions, as currently, CPU mode is simply not fast enough to compete with other projects such as VIBE.
